source/common/http/http1/codec_impl.h

```diff
@@ -241,6 +241,7 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
 
   Network::Connection& connection_;
   CodecStats& stats_;
+  const Http1Settings codec_settings_;
```

Comment:
I'm not quite sure about `settings` lifetime, could it be reference?

Reply:
I mean pass as a reference parameter, not store as a reference, otherwise you may copy twice. I think it may be safe to actually store as a reference but that would need to be analyzed and it is probably not worth it if the object is small.